server/util/url/url.go

```diff
@@ -0,0 +1,18 @@
+package url
+
+import (
+	"regexp"
+	"strings"
+)
+
+var (
+	userPasswordRegex = regexp.MustCompile("^(https?://)(.*?:.*?@)")
```

Comment:
i think we should probably support more formats, like git://, ssh://, and one without a prefix:
https://stackoverflow.com/questions/31801271/what-are-the-supported-git-url-formats

Reply:
or you can go straight to the source: https://github.com/git/git/blob/77bd3ea9f54f1584147b594abc04c26ca516d987/t/t0110-urlmatch-normalization.sh

Reply:
Now using the `git-urls` library for parsing, with some additional logic layered on top to handle protocol-less URLs (e.g. `github.com/foo/bar`) and `localhost`